### PR TITLE
Reorganize example gallery

### LINKED ISSUE #595: Example Gallery: Text and LaTeX --> Reference manual
See https://manimce.readthedocs.io/en/latest/examples/tex.html, this page currently contains much more than just a gallery of examples.

Guided explanations on how to use `Tex` and `Text`, respectively, should be located in the reference manual (I'd suggest the module docstrings of `tex_mobject.py` and `text_mobject.py`).

The example gallery should link prominently (details not fully clear yet, see #448) to the module documentation. As a first triage, this could be done with a SEEALSO-block.

### PR DESCRIPTION
**Pull request by behackl**
This PR is intended to enforce a stricter separation between examples / explanations that should be located in the reference manual vs. examples that should be contained in the example gallery.

In #591 we arrived the following distinction (credit to @leotrs for the formulations):

- Reference manual (docstrings of classes and methods): close to code, each example illustrates a single method/keyword. Stand alone examples.
- Reference manual (module docstrings): Overview over functionalities of an entire module; narrative is welcome but not required
- Example Gallery: close to human, little/no narrative, Stand alone examples. Each example illustrates more than one feature or class. (Note I'm using "feature" to mean "more than a single method/keyword". E.g being able to change latex fonts is a feature, but the actual keyword argument used to do so is not.)
- Tutorials: closer to human, mostly narrative. Examples build upon each other (not necessarily stand-alone).

This PR is work in progress and also intended to collect code changes and the corresponding discussions in one place.

Eventually, this should resolve #595 and maybe also #448. 

## Acknowledgement
- [x] I have read the [Contributing Guidelines](https://github.com/ManimCommunity/manim/wiki/Documentation-guidelines-(WIP))

### CODE REVIEW

**Comment by behackl** [comment]
As a first step, I am going over all examples currently included in the gallery -- and make a suggestion where I think they should live. In a second step, the remaining gallery examples should be reorganized. I'll try to make this as review friendly as possible. As there is a lot of moving, I suggest to look at the changes commit-by-commit.

Keep in mind that this is just a suggestion for the classification! There *definitely* will have to be more work to tie everything together nicely. Ultimately, I'd maybe even like to have a full (autogenerated) list of figures / rendered examples from our entire documentation... but baby steps.

### Shapes, Images and Positions
(Commit: 2940b85f)
- `ShowScreenResolution` --> `tutorials/configuration.rst`
- `GeometricShapes` --> Docstring of `Mobject.next_to`
- `PointMovingOnShapes`: keep.
- `GradientImageFromArray`: remove -- the (almost same) example is already in the docstring of `ImageMobject`.
- `ArcShapeIris` --> Docstring of `VGroup` (although I am definitely not married to neither the example or the target location).
- `DotInterpolation` --> Docstring of `Mobject.interpolate` (I *do* find it concerning that `dotMiddle` is not actually in the middle between `dotL` and `dotR`, we should fix that as well.)
- `MovingAround`: keep, although this could also live in the docstring of `Scene.play`.
- `TextAlignment` --> Module docstring of `text_mobject`.
- `BezierSpline`: keep, although I *do* think that this is an example with a lot of lines of code for what it does.

### Annotations
(Commit: e726b836)
- `BraceAnnotation`: keep.
- `VectorArrow`: keep.
- `UsefulAnnotations` --> Module docstring of `geometry`. (I am not 100% happy with this. I'd like to see some of the stuff in `geometry.py` moved into an `annotations` module or so, there it would fit even better.)
- `ArrowTipsShowcase` --> Docstring of  `ArrowTip`.

### Plotting with Manim
(Commit: bda45b2c)
- `FunctionPlot`: remove.
- `FunctionPlotWithLabbeledYAxe` --> Module docstring of `graph_scene`
- `SequencePlot` --> Docstring of `GraphScene.coords_to_point`
- `GaussianFunctionPlot1` --> Module docstring of `graph_scene`.
- `GaussianFunctionPlot2`: remove. I can remember the discussion around this, but I strongly feel that our documentation is *not* the place to illustrate basic Python concepts.
- `SinAndCosFunctionPlot`: keep.
- `GraphAreaPlot`: keep.
- `HeatDiagramPlot`: keep.
- `ParametricFunctionWithoutGraphScene` --> Docstring of `ParametricFunction`

### Text and LaTeX
(Commit: 82ffc977)
- I've moved the very comprehensive and structured description of Tex mobjects to the module docstring of `tex_mobject`.
- The less comprehensive description for Text is moved to the module docstring of `text_mobject`. It might make sense to relocate more of the examples in the docstring of `Text` to the module description, but let's do that in a second step.

### Formulas
(Commit: 64e8625b)
- `MovingFrameBox`: keep.
- `MovingBraces`: Moved to the module docstring of `tex_mobject` (into the comprehensive description) as a better example for `MathTex`.

### 3D Scenes
(Commit: 53cde4b2)
- `ThreeDSphere`: remove. (This is virtually the same example as `ThreeDLightSourcePosition` but with the default lighting position. I don't see a reason to keep both.)
- `ThreeDParametricSpring` --> Docstring of `ParametricFunction`
- Rest: keep.

(Taking a break now, will finish these suggestions for the remaining sections (Camera Settings, Animations, Advanced Projects) later.)

**Comment by leotrs** [comment]
Untitled Diagram

**Comment by behackl** [comment]
Second part of my suggestion:

### Camera settings
(Commit: b5d1cf88)
- `ChangingCameraWidth`: remove.
- `ChangingCameraWidthAndRestore`: module docstring of of `moving_camera_scene`
- `ChangingCameraWidthAndRevert`: remove.
- `MovingCameraCenter` --> module docstring of of `moving_camera_scene`
- `MovingAndZoomingCamera` --> module docstring of of `moving_camera_scene`
- `MovingCameraOnGraph` --> module docstring of of `moving_camera_scene`
- `FollowingGraphCamera`: keep.
- `UseZoomedScene` --> module docstring of `zoomed_scene`
- `ChangingZoomScale` --> module docstring of `zoomed_scene`
- `MovingZoomedSceneAround`: keep.

### Animations
(Commit: 7cac4986)
- `RotationUpdater`: remove.
- `RotationUpdater2` --> Docstring of `Mobject.add_updater`.
- `NumberLinePointer`: remove, this is a duplicate of the example in the docsstring of `ValueTracker`.
- `PointWithTrace`: keep.
- `PointMovingWithValTracker`: remove, I think the other example for `ValueTracker` is better.
- `RotationValTracker`: remove. This is too close to the RotationUpdater example.
- `PlaneFadeOut` --> Docstring of `FadeOut`.
- `FadeInAndOut`: remove.

### Advanced Projects
(Commit: ab93b0b9)
- `OpeningManim`: excellent stuff, keep.
- `SquareToCircle`: we already have this several times in the tutorial. remove.
- `WarpSquare` --> docstring of `ApplyPointwiseFunction`
- `WriteStuff`: remove, already covered by `OpeningManim`
- `SquareMovingWithUpdaters`  --> docstring of `DecimalNumber`
- `ShapesWithVDics` --> docstring of `VDict` (not sure what really happens in this example,  but it does a lot of stuff with VDict, so let's move it there)
- `VariablesWithValueTracker` --> docstring of `Variable`
- `SineCurvePlot`: excellent, keep.

**Comment by behackl** [comment]
@leotrs excellent flowchart, I like it a lot! :-)

**Comment by behackl** [comment]
Are there any comments regarding this proposed redistribution of examples, @ManimCommunity/docs?

If not, I'll go on to the second stage and rearrange the remaining examples in the gallery. I'm not entirely sure about how to do that yet, but I am also playing with the idea of rearranging everything in different sections of one single file.

**Comment by leotrs** [comment]
I would like to know what @kolibril13 thinks before moving forward. I believe @cobordism also has Opinions.

**Comment by kolibril13** [comment]
I will look on this today in the evening!

**Comment by kolibril13** [comment]
Comments part 1)
> ### Shapes, Images and Positions
> (Commit: [2940b85](https://github.com/ManimCommunity/manim/commit/2940b85ff44679fe717f77a038433e36c48f6102))
> 
> * `ShowScreenResolution` --> `tutorials/configuration.rst`
> * `GeometricShapes` --> Docstring of `Mobject.next_to`

Maybe we can make a new example where we show the 10 most  common shapes and add them here
> * `PointMovingOnShapes`: keep.
> * `GradientImageFromArray`: remove -- the (almost same) example is already in the docstring of `ImageMobject`.
> * `ArcShapeIris` --> Docstring of `VGroup` (although I am definitely not married to neither the example or the target location).
> * `DotInterpolation` --> Docstring of `Mobject.interpolate` (I _do_ find it concerning that `dotMiddle` is not actually in the middle between `dotL` and `dotR`, we should fix that as well.)

We can name the dot dot_interpol, I think that setting the parameter alpha=0.3 is quite convenient to quickly understand its usage.
> * `MovingAround`: keep, although this could also live in the docstring of `Scene.play`.
> * `TextAlignment` --> Module docstring of `text_mobject`.
> * `BezierSpline`: keep, although I _do_ think that this is an example with a lot of lines of code for what it does.

yep, maybe we can find a better way to make this. That was a project once of mine and I did not care about making this as short as possible.

Further,  the current master has some more examples:
https://docs.manim.community/en/latest/examples/shapes_images_positions.html


ManimCELogo -> keep
ManimCELogoFromSVG -> has to be fired, this looks very strange right now:
https://docs.manim.community/en/latest/examples/shapes_images_positions.html#manimcelogofromsvg
ManimCELogoFromImage-> should be scaled or removed.
https://docs.manim.community/en/latest/examples/shapes_images_positions.html#manimcelogofromimage

**Comment by behackl** [comment]
> Maybe we can make a new example where we show the 10 most common shapes and add them here
> 

That sounds like an excellent idea to me, I'd put it in the `geometry` module docstring, though. And something for a follow-up, I'd like to keep this PR about reorganization.

> We can name the dot dot_interpol, I think that setting the parameter alpha=0.3 is quite convenient to quickly understand its usage.
> 

With your comment, I've understood that also the position of the Mobjects is interpolated. The parameter alpha made me think that this is mainly about blending the objects together.

> yep, maybe we can find a better way to make this. That was a project once of mine and I did not care about making this as short as possible.
> 

We can look into it -- not on this PR, though.

> Further, the current master has some more examples:
> https://docs.manim.community/en/latest/examples/shapes_images_positions.html
> 

Argh. I merged master, not sure why they were not included on the branch I started. Anyways, I will update my comments above including these examples.

> ManimCELogo -> keep

+1, for now -- I'd like to replace it with @eulertour's logo/banner animation eventually.

> ManimCELogoFromSVG -> has to be fired, this looks very strange right now:
> https://docs.manim.community/en/latest/examples/shapes_images_positions.html#manimcelogofromsvg

+1 for removing. We can re-add it once parsing the SVG is not broken anymore.

> ManimCELogoFromImage-> should be scaled or removed.
> https://docs.manim.community/en/latest/examples/shapes_images_positions.html#manimcelogofromimage

+1 for removing.

**Comment by kolibril13** [comment]
> That sounds like an excellent idea to me, I'd put it in the `geometry` module docstring, though. And something for a follow-up, I'd like to keep this PR about reorganization.
> 
> >

Ok, I've added this into this issue here: https://github.com/ManimCommunity/manim/issues/591

**Comment by kolibril13** [comment]
And part 2 of my comments:
Note that when i did not write anything, I agree on the suggestion.
> ### Annotations
> (Commit: [e726b83](https://github.com/ManimCommunity/manim/commit/e726b836076ae6718d8a506df4b84aeba12221bf))
> 
> * `BraceAnnotation`: keep.
> * `VectorArrow`: keep.
> * `UsefulAnnotations` --> Module docstring of `geometry`. (I am not 100% happy with this. I'd like to see some of the stuff in `geometry.py` moved into an `annotations` module or so, there it would fit even better.)
Yeah, but as the 10 most common shapes, I would also like to have something like the 10 most common annotations here.
Or we make one giant example that includes **all** of these features and put it intro Shapes, Images and positions?
> * `ArrowTipsShowcase` --> Docstring of  `ArrowTip`.
ok, but then maybe we can add some arrows of these into this big example of annotations.
> ### Plotting with Manim
> (Commit: [bda45b2](https://github.com/ManimCommunity/manim/commit/bda45b2c51abc1715f00a8f6b7c4b968a7d8cf11))
> 
> * `FunctionPlot`: remove.
> * `FunctionPlotWithLabbeledYAxe` --> Module docstring of `graph_scene`
> * `SequencePlot` --> Docstring of `GraphScene.coords_to_point`
I would keep one example of this as well in one plot in the example gallery, as this is quite an essential tool for manim. 
E.g to SinAndCosFunctionPlot. Then we can remove SequencePlot
> * `GaussianFunctionPlot1` --> Module docstring of `graph_scene`.
> * `GaussianFunctionPlot2`: remove. I can remember the discussion around this, but I strongly feel that our documentation is _not_ the place to illustrate basic Python concepts.
ok,  we can remove this...
> * `SinAndCosFunctionPlot`: keep.
> * `GraphAreaPlot`: keep.
> * `HeatDiagramPlot`: keep.
> * `ParametricFunctionWithoutGraphScene` --> Docstring of `ParametricFunction`
good point!

> ### Text and LaTeX
> (Commit: [82ffc97](https://github.com/ManimCommunity/manim/commit/82ffc977f27fa47f0abae6dbcc43ae260542489c))
> 
> * I've moved the very comprehensive and structured description of Tex mobjects to the module docstring of `tex_mobject`.
> * The less comprehensive description for Text is moved to the module docstring of `text_mobject`. It might make sense to relocate more of the examples in the docstring of `Text` to the module description, but let's do that in a second step.
Sounds good!
> 
> ### Formulas
> (Commit: [64e8625](https://github.com/ManimCommunity/manim/commit/64e8625b4e411c786525997f4e28fd14963fb0f2))
> 
> * `MovingFrameBox`: keep.
> * `MovingBraces`: Moved to the module docstring of `tex_mobject` (into the comprehensive description) as a better example for `MathTex`.
Hmm, I would like to keep these two examples somehow together, as they are related to each other.
Also, I don't like the category "Formulas" with only one example not that much. Maybe we can get rid of this category completely and then concatenate these two scenes to only one and move them to annotations.
But moving to the module docstring of `tex_mobject` sounds good additionally.
> 
> ### 3D Scenes
> (Commit: [53cde4b](https://github.com/ManimCommunity/manim/commit/53cde4b2f57d760df253606a2ea6346437393927))
> 
> * `ThreeDSphere`: remove. (This is virtually the same example as `ThreeDLightSourcePosition` but with the default lighting position. I don't see a reason to keep both.)
With both examples, it will be easy to grasp where the initial light source is.
Buut when I think about it,  I would say that it does not matter that much.
Also, Another idea that I have right now is that the light source could be disabled by default.

> * `ThreeDParametricSpring` --> Docstring of `ParametricFunction`
sounds good!
> * Rest: keep.

**Comment by kolibril13** [comment]
The last part of my review will probably come tomorrow.
Maybe we can even have a meeting on discord and make these changes together?
That might be a bit easier than to plan it here in messages.
And another question:
Do we want to allow redundancies in the Reference and the Example section?

**Comment by behackl** [comment]
> > * `UsefulAnnotations` --> Module docstring of `geometry`. (I am not 100% happy with this. I'd like to see some of the stuff in `geometry.py` moved into an `annotations` module or so, there it would fit even better.)
> 
> Yeah, but as the 10 most common shapes, I would also like to have something like the 10 most common annotations here.
> Or we make one giant example that includes **all** of these features and put it intro Shapes, Images and positions?
> 

I agree with the idea. I suggest that we merge certain modules (`shape_matchers`, `brace`, parts of `geometry`) into a new module, `annotations` later on. Let us keep this PR to reorganizing the existing examples, however.

> > * `ArrowTipsShowcase` --> Docstring of  `ArrowTip`.
> 
> ok, but then maybe we can add some arrows of these into this big example of annotations.
> 

Arrows could also live in the annotations module mentioned above. I think I would like that.

> > * `SequencePlot` --> Docstring of `GraphScene.coords_to_point`
> 
> I would keep one example of this as well in one plot in the example gallery, as this is quite an essential tool for manim.
> E.g to SinAndCosFunctionPlot. Then we can remove SequencePlot
> 

The functionality of adding the dots is already documented in `HeatDiagramPlot`, which I would like to keep. This is why I propose to move this to the `coords_to_point` docstring.

> > * `GaussianFunctionPlot2`: remove. I can remember the discussion around this, but I strongly feel that our documentation is _not_ the place to illustrate basic Python concepts.
> 
> ok, we can remove this...
> 

Just to clarify my thoughts: I am not against such an example in principle, I just think that Python tutorials should handle explaining Python, and we should stick to explaining how our API works. Showing multiple ways of how to do something in the example gallery is the wrong location in my opinion: this could be done in a separate tutorial, where you also have some lines in between explaining the difference between the two variants -- but in the example gallery, it is unnecessary clutter. (I feel, the example gallery should be about getting an impression what Manim can do, and less about different ways of how to do it.) :-)

> > * `MovingBraces`: Moved to the module docstring of `tex_mobject` (into the comprehensive description) as a better example for `MathTex`.
> 
> Hmm, I would like to keep these two examples somehow together, as they are related to each other.
> Also, I don't like the category "Formulas" with only one example not that much. Maybe we can get rid of this category completely and then concatenate these two scenes to only one and move them to annotations.

I feel that both examples rather belong in the module docstring of the `annotations` module mentioned above; there they would fit nicely. I agree with your assessment that the "Formulas" section doesn't really fit; and I agree that we should simply get rid of it.

> > * `ThreeDSphere`: remove. (This is virtually the same example as `ThreeDLightSourcePosition` but with the default lighting position. I don't see a reason to keep both.)
> 
> With both examples, it will be easy to grasp where the initial light source is.
> Buut when I think about it, I would say that it does not matter that much.

Someone interested in 3D will be able to deduce that there is a light source and how its position can be changed from the second example; I don't quite see the value added by the first example. (I am also suggesting a rather aggressive approach for reorganizing the gallery; I think fewer but very diverse examples are better than many with repeating and/or only slightly changed outputs.)

Thanks for looking through all of this; as far as I can tell we agree on (almost) all suggestions made here, with several additional ideas for further examples to be added.

In the end, I'd like to add a paragraph of text to the landing page of the example gallery which explains that the gallery features a broad overview of how many different classes in Manim can be used together, and that more documentation concerning specific modules (especially geometry, plotting, tex, text...) including rendered examples can be found in the corresponding module documentation. This should help to connect the gallery back to other examples that might seem "lost in the documentation" currently.

**Comment by behackl** [comment]
> The last part of my review will probably come tomorrow.

Thank you for looking at all of this so carefully.

> Maybe we can even have a meeting on discord and make these changes together?
> That might be a bit easier than to plan it here in messages.

We can always move parts of the discussion to Discord, sure! :-)

> And another question:
> Do we want to allow redundancies in the Reference and the Example section?
> I am in favour of yes, because some examples are essential to manim that it would not feel right to hide them somewhere in the reference, as the example place is that place where people will get a good overview.

The reason I don't like redundancies for the examples is that the situation where an example occurs more than once in the documentation inevitably leads to inconsistencies sooner or later. I'd like to avoid them as much as possible; sometimes there might be a good reason for them, though.

I'd prefer to avoid the situation where we "hide them somewhere in the reference" as much as possible, and there might be a good way to do so: I think that all candidates for examples that could live in either the gallery as well as the reference manual would live in module docstrings in the reference manual. (I feel that examples for single classes/methods are, in general, too specific to show it in the gallery. And whether https://manimce--687.org.readthedocs.build/en/687/examples.html links to further example-subpages, or simply to module documentation like https://manimce--687.org.readthedocs.build/en/687/reference/manim.scene.graph_scene.html#module-manim.scene.graph_scene does not make that much of a difference, in my opinion.

**Comment by kolibril13** [comment]
Some quick thoughts on this:


> ### Animations
> (Commit: [7cac498](https://github.com/ManimCommunity/manim/commit/7cac498613b919abfb28370e17fcb0023bf67f1d))
> 
> * `RotationUpdater`: remove.

ok!
> * `RotationUpdater2` --> Docstring of `Mobject.add_updater`.

There should be at least one easy example with the use of add_updater with an explicit updater function (so no lambda function) in the animation section I think.
Also, there is a difference whether add_updater is used with or without the dt parameter, which I don't completely understand.

> * `NumberLinePointer`: remove, this is a duplicate of the example in the docsstring of `ValueTracker`.

agree, this can be removed.
> * `PointWithTrace`: keep.
> * `PointMovingWithValTracker`: remove, I think the other example for `ValueTracker` is better.

Agree on this as well.

> * `RotationValTracker`: remove. This is too close to the RotationUpdater example.

Same think I mentioned with the dt parameter up there, I think this should be clarified somewhere how this should be used.
```py
        def my_rotation_updater(mobj):
            mobj.rotate_about_origin(1 / 30) # be careful: This is framerate dependent!
        line_reference = Line(ORIGIN, LEFT).set_color(WHITE)
```
But when we get there some more clarity, I am fine with removing this.

> * `PlaneFadeOut` --> Docstring of `FadeOut`.

yeah, that sounds good!
> * `FadeInAndOut`: remove.
This example shows three features: FadeIn, FadeOut, and become.
Therefore, I would like to keep it, as this is not shown somewhere else in a so tight way.

**Comment by kolibril13** [comment]
> * `OpeningManim`: excellent stuff, keep.

Agreed! Maybe we can scale the first tex a bit with this line added:
        VGroup(title, basel).set_width(config["frame_width"] - 2 * LARGE_BUFF).arrange(DOWN)

> * `SquareToCircle`: we already have this several times in the tutorial. remove.

Yep!
> * `WarpSquare` --> docstring of `ApplyPointwiseFunction`

Good idea!
> * `WriteStuff`: remove, already covered by `OpeningManim`


> * `SquareMovingWithUpdaters`  --> docstring of `DecimalNumber`
> * `ShapesWithVDics` --> docstring of `VDict` (not sure what really happens in this example,  but it does a lot of stuff with VDict, so let's move it there)

I have never used VDics before, so I really don't mind about this one, so I agree on bury it there :) 
> * `VariablesWithValueTracker` --> docstring of `Variable`

That's nice!
> * `SineCurvePlot`: excellent, keep.
Agree!

**Comment by kolibril13** [comment]
> I'd prefer to avoid the situation where we "hide them somewhere in the reference" as much as possible, and there might be a good way to do so: I think that all candidates for examples that could live in either the gallery as well as the reference manual would live in module docstrings in the reference manual. (I feel that examples for single classes/methods are, in general, too specific to show it in the gallery. And whether https://manimce--687.org.readthedocs.build/en/687/examples.html links to further example-subpages, or simply to module documentation like https://manimce--687.org.readthedocs.build/en/687/reference/manim.scene.graph_scene.html#module-manim.scene.graph_scene does not make that much of a difference, in my opinion.


I think all functionalities should at least be touched once in the example gallery, for example VariablesWithValueTracker could come together with some other examples into the section of "Animations"

**Comment by kolibril13** [comment]
> ### Camera settings
> (Commit: [b5d1cf8](https://github.com/ManimCommunity/manim/commit/b5d1cf88e893043a19af47b8a3678f33f64ee20e))
> 
> * `ChangingCameraWidth`: remove.

agree
> * `ChangingCameraWidthAndRestore`: module docstring of of `moving_camera_scene`

Yeah, I think this should be seen in moving_camera_scene, however, I think it is quite important to have it here in the example section as well.
> * `ChangingCameraWidthAndRevert`: remove.

Same here, I think we should keep it, as it shows a case that can be used quite often.
> * `MovingCameraCenter` --> module docstring of of `moving_camera_scene`

hmm, I would also like to still have this here, and a redundancy in moving_camera_scene
> * `MovingAndZoomingCamera` --> module docstring of of `moving_camera_scene`
> * `MovingCameraOnGraph` --> module docstring of of `moving_camera_scene`
> * `FollowingGraphCamera`: keep.
> * `UseZoomedScene` --> module docstring of `zoomed_scene`
> * `ChangingZoomScale` --> module docstring of `zoomed_scene`
> * `MovingZoomedSceneAround`: keep.

I am not that happy that https://manimce--687.org.readthedocs.build/en/687/examples/camera_settings.html
would only have 2 examples remaining here.
I see that it makes sense to move many of them to moving_camera_scene, however, when I come new to manim and want to use this camera feature, I think I would love to have an overview on only one page in the docs of all possibilities.
Further, when I am more experienced in manim, I would love to have some tight examples at moving_camera_scene.
So this is a case, where I would find it ok to have redundancies.

**Comment by leotrs** [comment]
> I am not that happy that https://manimce--687.org.readthedocs.build/en/687/examples/camera_settings.html
would only have 2 examples remaining here.

I am personally fine with leaving some pages with few examples, especially if moving the current examples is according to (some version of) the flow diagram posted above.

If this moving leaves some pages with few examples, I feel like the solution is to add new ones, rather than leave them there.

The point is that we are just now beginning to build the examples section and *a lot* of the pages have fewer examples than the should, and we have fewer pages than we should overall as well.

Perhaps adding a short line "this page is under construction, check back later" or even "for more examples, see the reference section of such and such class" might be a good temporary compromise?

**Comment by leotrs** [comment]
Alternatively, we can flag this page for cleanup once we have more examples overall.

**Comment by behackl** [comment]
I've communicated this to @kolibril13 in private already, but I see that I have missed to make my ultimate intention here very clear. Practically, I would like to move the examples hosted as subpages of https://docs.manim.community/en/latest/examples.html directly to that page, which should be a well-curated selection of high-quality and best practice examples of what Manim is capable of.

To be more precise, I'd like to add a statement like
```
This gallery contains a collection of best practice code snippets together with their corresponding video/image output, illustrating different functionalities all across the library. Enjoy this taste of Manim!

Note, however, that this is not the only place in our documentation where you can see explicit code and video examples: there are many more in our reference manual -- see, for example, our documentation for the modules [insert list of important/well-documented modules in the reference manual].
```
aside from adding references to relevant classes, modules and functions to all examples in the gallery. This also strengthens the connection between the gallery and the reference manual (and keeps the examples that we have currently moved to the module docs within exactly the same reach from from https://docs.manim.community/en/latest/examples.html as they were).

(Even though the page would be somewhat long with many examples, I don't think that this is a bad thing -- on the contrary, if we structure the examples in, like, 3-5 categories and provide anchors, we can still have an easy-to-navigate page while also allowing users to see examples immediately, which is very important, psychologically. I don't think hitting users with a list of categories to choose from as we do now is beneficial, they are more likely to discover the full set of Manim's capabilities if everything is in one place.)

**Comment by leotrs** [comment]
That sounds good, but it is also a departure from the original intent of the examples gallery, no? Or maybe I'm misunderstood from the beginning.

**Comment by behackl** [comment]
I am not entirely sure; I think the main intention behind the gallery (showcasing the general capabilities of Manim) remains the central focus. If the original intention was to collect a variety of very simple examples illustrating each and every feature that Manim has to offer then yes: I simply don't think this approach can keep up with the speed Manim is developed at for very long.

And of course, just because I started moving stuff around, this PR is of course meant as a suggestion. It is the result of trying to create a system that scales well and requires as little regular maintenance as possible, tied together with some arguments why I think that this is a good solution -- but by no means should this be seen as a final decision. :-)

**Comment by behackl** [comment]
In d309c70c I have moved the examples from the subpages to `examples.rst` and added both a local TOC and the introductory paragraph to the gallery.

In my opinion, this single page collection still works very well, I'm happy to hear your opinions.

Remaining tasks on this PR:
- [ ] Settle with @kolibril13 which examples should be reintroduced to the gallery
- [ ] Add references for classes/functions/modules to all examples in the gallery
- [ ] Rearrange the existing categories in a way that separates them a bit better from each other

My suggestion for the third point is:
- Basic Concepts (= Shapes, Images and Positions + Annotations + Formulas)
- Plotting with Manim (= Plotting with Manim)
- Special Camera Settings (= Camera Settings + 3D)
- Advanced Projects (= Advanced Projects)

Any thoughts?

**Comment by behackl** [comment]
adb129e7 adds references for relevant functions and classes. References to most functions do not link properly because -- you guess correctly -- these functions do not have docstrings. We should add basic docstrings to all of them, but I'd like to delegate this to a follow-up PR.

**Comment by kolibril13** [comment]
> Perhaps adding a short line "this page is under construction, check back later" or even "for more examples, see the reference section of such and such class" might be a good temporary compromise?

"this page is under construction, contributions are more than welcome and check back later." maybe.  

> * Basic Concepts (= Shapes, Images and Positions + Annotations + Formulas)
> * Plotting with Manim (= Plotting with Manim)
> * Special Camera Settings (= Camera Settings + 3D)
> * Advanced Projects (= Advanced Projects)

I would add one more category "Animations" (=Write + FadeIn + Transform + become + updater)

**Comment by behackl** [comment]
> > Perhaps adding a short line "this page is under construction, check back later" or even "for more examples, see the reference section of such and such class" might be a good temporary compromise?
> 
> "this page is under construction, contributions are more than welcome and check back later." maybe.

My suggestion: let's wait and see how the page looks like when we are done with it; maybe it won't look so "under construction" in the end. :-)

> > * Basic Concepts (= Shapes, Images and Positions + Annotations + Formulas)
> > * Plotting with Manim (= Plotting with Manim)
> > * Special Camera Settings (= Camera Settings + 3D)
> > * Advanced Projects (= Advanced Projects)
> 
> I would add one more category "Animations" (=Write + FadeIn + Transform + become + updater)

Good call! Order? Something like this?

- Basic Concepts
- Animations
- Plotting with Manim
- Special Camera Settings
- Advanced Projects

**Comment by kolibril13** [comment]
> My suggestion: let's wait and see how the page looks like when we are done with it; maybe it won't look so "under construction" in the end. :-)
> 
> > >

ok! 

> > > Perhaps adding a short line "this page is under construction, check back later" or even "for more examples, see the reference section of such and such class" might be a good temporary compromise?
> > 
> > 
> > "this page is under construction, contributions are more than welcome and check back later." maybe.
> 
> My suggestion: let's wait and see how the page looks like when we are done with it; maybe it won't look so "under construction" in the end. :-)
> 
> > > * Basic Concepts (= Shapes, Images and Positions + Annotations + Formulas)
> > > * Plotting with Manim (= Plotting with Manim)
> > > * Special Camera Settings (= Camera Settings + 3D)
> > > * Advanced Projects (= Advanced Projects)
> > 
> > 
> > I would add one more category "Animations" (=Write + FadeIn + Transform + become + updater)
> 
> Good call! Order? Something like this?
> 
> * Basic Concepts
> * Animations
> * Plotting with Manim
> * Special Camera Settings
> * Advanced Projects

Yes!

**Comment by kolibril13** [comment]
@leotrs :
Can you provide the link to your amazing flowchart for us?
https://github.com/ManimCommunity/manim/pull/687#issuecomment-723625860
It would be really cool if it could be directly embedded from the site where it was created into the docs.
Otherwise, maybe we can embed it as an svg file?
I would prefer svg over png, as pngs have much bigger file size and at the same time less quality.

**Comment by leotrs** [comment]
https://drive.google.com/file/d/1aKyJTloYB97IhrzwqEENOu-WQyuVWMjM/view?usp=sharing

**Comment by behackl** [comment]
We have not talked about bringing back examples from other locations yet, but I feel with the current solution that links explicitly to our most illustrative module descriptions already included here, we could also delegate this to a follow-up PR. This is getting rather large and messy. (I've also coordinated with @kolibril13 about this.)

My suggestion for reordering the current gallery examples came in via ba233f07, I think this is ready for review.

**Review by kolibril13 — APPROVED** [review]

**Comment by kolibril13** [comment]
Nice!
I think I still have to get used to the idea of only having one large example file, but I can definitely see some benefits.
Just had the idea that we can create a post into the discord announcement channel as soon as this is merged into the master, saying something like:
"""We reorganized the example chapter in the docs, all manim interested people are more than welcome to give feedback
to the #docs channel on how you find this structure and your ideas to make it an even better user experience.
"""

**Comment by behackl** [comment]
> we can create a post into the discord announcement channel as soon as this is merged

Great idea!

**Review by eulertour — APPROVED** [review]

File: docs/source/examples/shapes_images_positions.rst
Comment (kolibril13) [inline docs/source/examples/shapes_images_positions.rst:79]: docs source

Comment (leotrs) [reply docs/source/examples/shapes_images_positions.rst:79]: @kolibril13 this was merged back in November - and I don't understand your comment here. Care to explain here?

Comment (kolibril13) [reply docs/source/examples/shapes_images_positions.rst:79]: I needed to reference this line in pr #926 , and making a comment on this line was the fastest way to do so ;)

**Pull request merged**
